web-api/src/server.rs

```diff
@@ -53,21 +61,21 @@ pub trait Application {
 
     /// Create an application specific extension to app state.
     //Design Note: We could handle this by adding `TyFrom<&Config<..>>` bounds
-    //             to `AppStateExtension` but using this helper method is simpler
+    //             to `Extension` but using this helper method is simpler
     //             and it is also easier to add async if needed (using #[async-trait]).
-    fn create_app_state_extension(
-        config: &Self::Config,
-    ) -> Result<Self::AppStateExtension, SetupError>;
+    fn create_extension(config: &Self::Config) -> Result<Self::Extension, ApplicationError>;
+
+    async fn setup_storage(config: &storage::Config) -> Result<Self::Storage, ApplicationError>;
 }
 
-pub type SetupError = anyhow::Error;
+pub(crate) type ApplicationError = anyhow::Error;
```

Comment:
I'm not sure why you changed the name.

It's not an application error (as in it happens during the runtime of the application), it's an error which basically only can happen during startup (and in very very unlikely cases shutdown).

Reply:
i renamed it because it's mainly the error which is returned in the `Application` trait methods, which seems more fitting. i'll revert it though.